web/src/flow/stages/saml_logout/SAMLLogoutStage.ts

```diff
@@ -0,0 +1,113 @@
+import "#flow/components/ak-flow-card";
+
+import { BaseStage } from "#flow/stages/base";
+
+import { FlowChallengeResponseRequest, SAMLLogoutChallenge } from "@goauthentik/api";
+
+import { msg, str } from "@lit/localize";
+import { CSSResult, html, nothing, PropertyValues, TemplateResult } from "lit";
+import { customElement } from "lit/decorators.js";
+import { ifDefined } from "lit/directives/if-defined.js";
+import { createRef, ref, Ref } from "lit/directives/ref.js";
+
+import PFButton from "@patternfly/patternfly/components/Button/button.css";
+import PFForm from "@patternfly/patternfly/components/Form/form.css";
+import PFFormControl from "@patternfly/patternfly/components/FormControl/form-control.css";
+import PFLogin from "@patternfly/patternfly/components/Login/login.css";
+import PFTitle from "@patternfly/patternfly/components/Title/title.css";
+import PFBase from "@patternfly/patternfly/patternfly-base.css";
+
+@customElement("ak-stage-saml-logout")
+export class SAMLLogoutStage extends BaseStage<SAMLLogoutChallenge, FlowChallengeResponseRequest> {
+    #formRef: Ref<HTMLFormElement> = createRef();
+
+    public static styles: CSSResult[] = [PFBase, PFLogin, PFForm, PFButton, PFFormControl, PFTitle];
+
+    public override firstUpdated(changedProperties: PropertyValues): void {
+        super.firstUpdated(changedProperties);
+
+        // If complete, auto-submit to continue flow
+        if (this.challenge.isComplete === "true") {
+            const submitEvent = new SubmitEvent("submit");
+            this.submitForm(submitEvent);
+            return;
+        }
+
+        // If POST binding, auto-submit the form
+        if (this.challenge.binding === "post" && this.#formRef.value) {
+            this.#formRef.value.submit();
+        }
+
+        // If redirect binding, perform the redirect
+        if (this.challenge.binding === "redirect") {
+            if (!this.challenge.redirectUrl) {
+                throw new TypeError(`Binding challenge does not a have a redirect URL`);
+            }
+            requestAnimationFrame(() => window.location.assign(this.challenge.redirectUrl!));
+        }
+    }
+
+    render(): TemplateResult {
+        let providerName = "SAML Provider";
+        const prefix = "Provider for ";
+        if (this.challenge.providerName && this.challenge.providerName.startsWith(prefix)) {
+            providerName =
+                this.challenge.providerName.slice(prefix.length) || this.challenge.providerName;
```

Comment:
We shouldn't modify strings that need to be translated like this, instead add a field to the challenge with just the provider name and then wrap it in msg("") here